Sources/MapboxMaps/Foundation/Camera/CameraManager.swift

```diff
@@ -174,122 +173,109 @@ public class CameraManager {
     }
 
     /// Moves the viewpoint to a different location using a transition animation that
     /// evokes powered flight and an optional transition duration and timing function
     /// It seamlessly incorporates zooming and panning to help
     /// the user find his or her bearings even after traversing a great distance.
     ///
-    /// NOTE: Keep in mind the lifecycle of the `CameraAnimator` returned by this method.
-    /// If a `CameraAnimator` is destroyed, before the animation is finished,
-    /// the animation will be interrupted and completion handlers will be called.
-    ///
     /// - Parameters:
     ///   - camera: The camera options at the end of the animation. Any camera parameters that are nil will not be animated.
     ///   - duration: Duration of the animation, measured in seconds. If nil, a suitable calculated duration is used.
     ///   - completion: Completion handler called when the animation stops
-    /// - Returns: The optional `CameraAnimator` that will execute the FlyTo animation
+    /// - Returns: An instance of `CameraAnimatorProtocol` which can be interrupted if necessary
+    @discardableResult
     public func fly(to camera: CameraOptions,
                     duration: TimeInterval? = nil,
-                    completion: AnimationCompletion? = nil) -> CameraAnimator? {
+                    completion: AnimationCompletion? = nil) -> CameraAnimatorProtocol? {
 
         guard let mapView = mapView else {
             return nil
         }
 
-        // Stop the `internalCameraAnimator` before beginning a `flyTo`
-        internalCameraAnimator?.stopAnimation()
-
-        guard let interpolator = FlyToInterpolator(from: mapView.cameraOptions,
-                                                   to: camera,
-                                                   size: mapView.bounds.size) else {
-            return nil
-        }
-
-        // If there was no duration specified, use a default
-        let time: TimeInterval = duration ?? interpolator.duration()
+        // Stop the `internalAnimator` before beginning a `flyTo`
+        internalAnimator?.stopAnimation()
 
-        // TODO: Consider timesteps based on the flyTo curve, for example, it would be beneficial to have a higher
-        // density of time steps at towards the start and end of the animation to avoid jiggling.
-        let timeSteps = stride(from: 0.0, through: 1.0, by: 0.025)
-        let keyTimes: [Double] = Array(timeSteps)
+        let flyToAnimator = FlyToAnimator(delegate: self)
+        mapView.cameraAnimatorsHashTable.add(flyToAnimator)
 
-        let animator = makeCameraAnimator(duration: time, curve: .linear) {
+        flyToAnimator.makeFlyToInterpolator(from: mapView.cameraOptions,
+                                             to: camera,
+                                             duration: duration,
+                                             screenFullSize: mapView.bounds.size)
```

Comment:
this could be replace by `mapboxMap.size` eventually if we want to try to further decouple CameraManager from knowing about a view

Reply:
not for this PR